docs/decisions/0007-executive-education-2U-tos-profile-data-route.md

```diff
@@ -0,0 +1,59 @@
+
+## Decision
+
+Users will access a new page route through the Executive Education (2U) allocation and fulfillment flow in order to view collect ToS and profile data from users, if applicable. 
+
+The enterprise-catalog service will modify its `enrollment_url` for Executive Education (2U) courses to point to a URL in the ecommerce service. This ecommerce URL will determine if the user needs to be redirected to this ToS and user profile data collection page (i.e., if they are not yet enrolled), and if so, ecommerce will redirect the user to this new route in the Learner Portal. Users will view and agree to the ToS and provide the required information in a form, and ultimately send this data as its payload back to the ecommerce POST URL, which would then handle the GEAG fulfillment via the `/allocations` API endpoint and redirect to the order history page.
+
+The content of the terms themselves will be retrieved from the `/terms` API endpoint in GEAG via a backend-for-frontend (BFF) endpoint. Similarly, should we need to display metadata about the Executive Education (2U) course itself (e.g., title, description, etc.), we will also source these data from a similar BFF endpoint to return product metadata in the UI.
+
+If the user does not accept the ToS, they should see appropriate messaging and provide a link to redirect back to their external LMS (i.e., where they came from). This URL will be provided by the ecommerce view that redirects the user to this new page route, including a redirect URL via a query parameter.
+
+If the user opts to decline ToS, we will inform the user that they must accept the ToS in order to continue with their enrollment, and provide a link to bring the user back to their external LMS.
+
+If the enterprise customer associated with the user does not have enough remaining balance on their offer, the user will be redirected to a route in the Learner Portal and see messaging that their organization no longer has enough remaining balance and to contact their organization's edX administrator. A link will be provided to bring the user back to their external LMS.
+
+The ecommerce view will provide the redirect URL and any failure reasons as query parameters to this new page route in the Learner Portal.
+
+In the event that the associated enterprise customer has the Executive Education (2U) configuration flag turned off, the new page route implemented via this ADR will return a 404 page instead. By doing so, we eliminate the need for a separate feature flag for the new page route.
+
+While on this page route, we will ensure the header does not include navigation links to the "Dashboard" or "Find a Course" page, as this user flow does not intend for these pages to be reachable, nor would they have the necessary data to provide a good user experience (i.e., Executive Education (2U course enrollments won't appear on the "Dashboard" or appear in the search results).
```

Comment:
A lot of the things you are describing here are also part of those sequence diagrams Chris wrote yesterday. What do you think about adding this as part of this ADR?